pkg/provisioner/crdprovisioner.go

```diff
@@ -641,12 +573,12 @@ func (c *CrdProvisioner) PublishVolume(
 		}
 	}
 
-	for i := 0; requiredUnpublishCount > 0 && i < len(unpublishOrder); i++ {
-		if err = c.detachVolume(ctx, unpublishOrder[i]); err != nil {
-			err = status.Errorf(codes.Internal, "failed to make request to unpublish volume (%s) from node (%s): %v", unpublishOrder[i].Spec.VolumeName, unpublishOrder[i].Spec.NodeName, err)
+	for _, replicaToDetach := range replicasToDetach {
+		w.Logger().V(5).Infof("Detaching replica attachment (%s) to make resource for primary attachment (%s)", replicaToDetach.Name, attachmentName)
+		if err = c.detachVolume(ctx, replicaToDetach); err != nil {
+			err = status.Errorf(codes.Internal, "failed to make request to unpublish volume (%s) from node (%s): %v", replicaToDetach.Spec.VolumeName, replicaToDetach.Spec.NodeName, err)
 			return nil, err
```

Comment:
Shouldn't we try to detach the next replica instead of just failing? The `detachVolume` function synchronously detaches the volume. Can there ever be more than one replica being detached? (`getReplicasToDetach` returns an array, so I assume yes.) If so, it would be good to do the detaches in parallel and wait for completion of all. This would reduce publish latency and increase the likelihood of the detaches happening in a single batch since they should be on the same node.